new-pipeline/src/crash-summary.js

```diff
@@ -0,0 +1,590 @@
+  d3.select("#detail-pluginhangs").text(kCommaFormat(d.pluginhangs));
+  d3.select("#detail-pluginhang-rate").text(k2Format(d.npapi_hangs_per_khour));
+  d3.select("#detail-gmpluginaborts").text(kCommaFormat(d.abortsgmplugin));
+  d3.select("#detail-gmpluginaborts-rate").text(k2Format(d.gmp_aborts_per_khour));
+  d3.select("#detail-gmpluginaborts").text(kCommaFormat(d.abortscontent));
+  d3.select("#detail-gmplugincrashes").text(kCommaFormat(d.crashesdetectedcontent));
+  d3.select("#detail-gmpluginaborts-rate").text(k2Format(d.content_aborts_per_khour));
+  d3.select("#detail-gmplugincrash-rate").text(k2Format(d.content_crashes_per_khour));
+  d3.select("#detail-plugin-submission-rate").text(d3.format(".2%")(d.plugin_crash_submit_rate));
+
+  d3.select("#detail-contentaborts").text(kCommaFormat(d.abortscontent));
+  d3.select("#detail-contentcrashes").text(kCommaFormat(d.crashesdetectedcontent));
+  d3.select("#detail-contentaborts-rate").text(k2Format(d.content_aborts_per_khour));
+  d3.select("#detail-contentcrashes-rate").text(k2Format(d.content_crashes_per_khour));
+  d3.select("#detail-content-detection-rate").text(kPctFormat(d.content_crash_detect_rate));
+  d3.select("#detail-content-submission-rate").text(kPctFormat(d.content_crash_submit_rate));
+
+  let details = $("#hover-details");
+  let width = parseInt(details.css("width"));
+  let height = parseInt(details.css("height"));
+
+  let [x, y] = d3.mouse(document.documentElement);
+  let position = { left: x - width - 5, top: y - height - 5 };
+  if (position.left < 5) {
+    position.left = x + 5;
+  }
+  if (position.top < 5) {
+    position.top = y + 5;
+  }
+  details.offset(position).toggleClass("invisible", false);
+}
+function do_mouseout() {
+  d3.select("#hover-details").classed("invisible", true);
+}
+
+function graph_it() {
+  gChannelDimension.filter(gChannel);
+
+  let group;
+  switch (gSort) {
+  case "buildid":
+    group = gBuildIDDimension.group();
+    break;
+  case "date":
+    group = gDateDimension.group();
+    break;
+  default:
+    throw new Error("Unexpected gSort");
+  }
+
+  let raw = group.reduce(counts_add, counts_sub, counts_initial)
+    .orderNatural().all();
+
+  let data = raw
+    .map((d) => (new BuildData(d, gSort == "buildid")))
+    .filter((d) => (d.date != null && d.subsessionlengths > 500000));
+  window.gGraphData = data;
+
+  // HOURS
+
+  let max_x = new Date();
+
+  MG.data_graphic({
+    title: "Usage Hours",
+    description: "Total usage hours: sum of main-pings payload.subsessionlength converted to hours",
+    data: data,
+    full_width: true,
+    height: 200,
+    target: '#hours-nightly',
+    x_accessor: 'date',
+    y_accessor: 'subsessionlengths',
+    max_x: max_x,
+    min_x: gMinDate,
+    utc_time: true,
+    mouseover: do_mouseover,
+    mouseout: do_mouseout,
+    interpolate: 'step',
+    linked: true,
+    linked_format: '%Y%m%d%H%M%S',
+    show_rollover_text: false,
+    area: false,
+  });
+
+  // MAIN CRASHES
```

Comment:
nit: These comments appear to be falling behind the code. MG's object format is verbose enough you could just omit.